Main.hs

```diff
@@ -11,28 +11,33 @@ import qualified Data.List as L
 import Text.Printf (printf)
 
 multByIndex :: [Int] -> [Int]
-multByIndex = undefined
+multByIndex xs = zipWith (*) xs [0..]
 
 powerByIndex :: [Int] -> [Int]
-powerByIndex = undefined
+powerByIndex xs = zipWith (^) xs [0..]
 
 productOfDifference :: [Int] -> Int
-productOfDifference = undefined
-
+productOfDifference xs
+  | length (take 3 xs) < 2 = error "undefined"
+  | otherwise = foldl (*) 1 (zipWith (-) xs (drop 1 xs))
+                            
 isSorted :: [Int] -> Bool 
-isSorted xs = undefined
+isSorted xs
+  | length (take 2 xs)  < 2 = True
+  | otherwise = foldl (\y (a,b) ->  y && (b >= a)) True (zip xs (tail xs))
```

Comment:
You have repeating code here in the guards which has slight variation in the constants, which surprises me. If you really want to stick with this style of checking whether the length of the list is big enough, I'd introduced a `isShort` predicate parameterized by the expected length of the list. Otherwise, just pattern matching on the empty list is enough for these to functions.